### PR TITLE
Change allowQueryParamWithNoEqual default to true for Servlet 5.0+

### LINKED ISSUE #16611: Change allowQueryParamWithNoEqual to true for servlet-5.0 and higher
Needed for recent TCK failure. See Thread Here: https://ibm-cloud.slack.com/archives/C01GSPHPSQ1/p1617657245002400 

PR Demonstration Failure: https://github.com/OpenLiberty/open-liberty/pull/16518 with additional background information on the failure:

`
The TCK sends content like "ABC", but when the JAX-RS resource method is invoked, the entity parameter is injected with "". In our testing, if we send content of "ABC=", then the injected entity parameter is "ABC". `

`
The reason for this is that RESTEasy uses the HttpServletRequest.getParameterMap() to determine the content when handling application/x-www-form-urlencoded requests. By default, the web container does not process parameters that do not contain an equals sign. Setting the allowqueryparamwithnoequal attribute to true fixes this.
`

_________

**Property:**  allowQueryParamWithNoEqual

**Description:** 
`When this property is set to true, if the query parameter in a URL contains only the string name, the server returns an empty string as the value for the request.getParameter(name) query. When this property is set to false, the server returns null as the value for the request.getParameter(name) query. The default value is false.`

To resolve the TCK failure, the allowQueryParamWithNoEqual property should set to true.  The alternative is to swap out the existing HashTable data structure to something that allows nulls -- tedious change with additional attention to synchronization.  

Additionally, other than the TCK failure, the open liberty default behavior doesn't match other servers.  Changing the default to true would let us match behaviors by returning empty strings.  

If a query doesn't have an equals sign, then it is dropped and not added to the parameter data map (not even the key) since [HashTables](https://docs.oracle.com/javase/8/docs/api/java/util/Hashtable.html) do not allow null values.  `allowQueryParamWithNoEqual` is a workaround by adding an empty string, not a null value. 

The key lines of code are here: https://github.com/OpenLiberty/open-liberty/blob/43eeef190f08cb845850d8bad1b21ab3c88d5b54/dev/com.ibm.ws.webcontainer/src/com/ibm/wsspi/webcontainer/util/RequestUtils.java#L541
___________
**Testing:** 

Test Servlet Code: https://gist.github.com/volosied/e8a45802114c4785293aeda2a5e39175 

I tested both POST & GET requests. 

URL: http://localhost:8080/servlet/HelloWorld?name=vlad&test  (Note: test has no equals) 

Open Liberty Current Behavior: 
```
Parameter Map: class java.util.Hashtable
_________________
getParameterMap
Key = name, Value = vlad // ---- no key test 
_________________
getParameter(test)
Key = test, Value = null  // -------  null instead of an empty string
```

Open Liberty Current Behavior (allowQueryParamWithNoEqual=“true”)
```
Parameter Map: class java.util.Hashtable
_________________
getParameterMap
Key = name, Value = vlad
Key = test, Value =
_________________
getParameter(test)
Key = test, Value =  // Empty String 

```
Tomcat 9.0.45, Glassfish 5.1, Jetty 
```
Parameter Map: // Varies for each server 
_________________
getParameterMap 
Key = name, Value = vlad
Key = test, Value =
_________________
getParameter(test)
Key = test, Value =  // Empty String 
```

The servers use different underlying Map data structures as indicated below: 

Tomcat 9.0.45, Glassfish 5.1 
```
Parameter Map: class org.apache.catalina.util.ParameterMap
```

Jetty 
```
Parameter Map: class java.util.Collections$UnmodifiableMap
```

Additionally, I ran a the full FAT buckets ( set to true for _all_ servlet versions) and all our tests passed.  Se [PR 1653](https://github.com/OpenLiberty/open-liberty/pull/16531)1 / [Build](https://wasrtc.hursley.ibm.com:9443/jazz/web/projects/WS-CD#action=com.ibm.team.build.viewResult&id=_MiEFIJlqEeumwuCQm6DXsQ).   Although testing was done for all serlvet versions, the actual PR would only be for servlet 5.0

### PR DESCRIPTION
**Pull request by volosied**
fixes #16611

Change has been approved in the POC

### CODE REVIEW

**Comment by volosied** [comment]
#build

**Review by helyarp — COMMENTED** [review]
Changes look good!

**Comment by volosied** [comment]
#build #spawn.fullfat.buckets=com.ibm.ws.webcontainer.servlet.4.0_fat,com.ibm.ws.webcontainer.servlet.3.1_fat,com.ibm.ws.webcontainer.servlet.4.0_fat_toleration.3.0,com.ibm.ws.webcontainer.servlet.4.0_fat_toleration.3.0.lWAS,com.ibm.ws.webcontainer_fat_servlet31_webcontainerlWAS,com.ibm.ws.webcontainer_fat_servlet31toleration

**Comment by volosied** [comment]
Unit tests fail as this scenario is not accounted for in the testing. Will update unit test to reflect new behavior.


com.ibm.ws.webcontainer.util.RequestUtilsTest > parseStringQueryStringUTF8 FAILED
    junit.framework.AssertionFailedError at RequestUtilsTest.java:47

com.ibm.ws.webcontainer.util.RequestUtilsTest > parseStringQueryStringISO8859_1 FAILED
    junit.framework.AssertionFailedError at RequestUtilsTest.java:47

com.ibm.ws.webcontainer.util.RequestUtilsTest > parseCharArrayQueryStringUTF8 FAILED
    junit.framework.AssertionFailedError at RequestUtilsTest.java:47

com.ibm.ws.webcontainer.util.RequestUtilsTest > parseCharArrayQueryStringISO8859_1 FAILED
    junit.framework.AssertionFailedError at RequestUtilsTest.java:47

**Comment by volosied** [comment]
#build #spawn.fullfat.buckets=com.ibm.ws.webcontainer.servlet.4.0_fat,com.ibm.ws.webcontainer.servlet.3.1_fat,com.ibm.ws.webcontainer.servlet.4.0_fat_toleration.3.0,com.ibm.ws.webcontainer.servlet.4.0_fat_toleration.3.0.lWAS,com.ibm.ws.webcontainer_fat_servlet31_webcontainerlWAS,com.ibm.ws.webcontainer_fat_servlet31toleration
@LibbyBot

**Comment by skasund** [comment]
L2 message review not required

**Comment by volosied** [comment]
#build #spawn.fullfat.buckets=com.ibm.ws.webcontainer.servlet.4.0_fat,com.ibm.ws.webcontainer.servlet.3.1_fat

File: dev/com.ibm.ws.webcontainer.servlet.4.0_fat/fat/src/com/ibm/ws/fat/wc/tests/WC5JakartaServletTest.java [hunk 1]
```diff
@@ -95,4 +95,32 @@ public void testSimple_Servlet50() throws Exception {
             }
         }
     }
+
+    /**
+     * Request a simple snoop servlet.
```
Comment (volosied) [inline dev/com.ibm.ws.webcontainer.servlet.4.0_fat/fat/src/com/ibm/ws/fat/wc/tests/WC5JakartaServletTest.java]: remove

**Review by pmd1nh — APPROVED** [review]

**Comment by volosied** [comment]
#build #spawn.fullfat.buckets=com.ibm.ws.webcontainer.servlet.4.0_fat,com.ibm.ws.webcontainer.servlet.3.1_fat

**Review by andymc12 — APPROVED** [review]
LGTM - thanks @volosied!

**Comment by volosied** [comment]
#run-libby-bot

**Pull request merged**